code/controllers/subsystem/fluids/fluid_effects.dm

```diff
@@ -0,0 +1,83 @@
+/**
+ *
+ */
+SUBSYSTEM_DEF(fluid_effect)
+	name = "Fluid Effect"
+	wait = 1
+	flags = SS_BACKGROUND|SS_POST_FIRE_TIMING
+
+	//
+	/// The amount of time
+	var/effect_wait = 1 SECONDS
+	///
+	var/list/list/obj/effect/particle_effect/fluid/processing
+	///
+	var/tmp/list/obj/effect/particle_effect/fluid/currentrun
+	///
+	var/tmp/num_buckets
```

Comment:
Why are you using /tmp?

Reply:
Because I felt like it.

Also, they will all be generated at init and I _think_ that's what tmp is for?